pkg/controllers/provisioning/scheduling/scheduler.go

```diff
@@ -469,6 +471,22 @@ func getDaemonOverhead(nodeClaimTemplates []*NodeClaimTemplate, daemonSetPods []
 	})
 }
 
+// getDaemonHostPortUsage tracks requested host ports for daemonsets, given a nodeclaimtemplate
+func getDaemonHostPortUsage(nodeClaimTemplates []*NodeClaimTemplate, daemonSetPods []*corev1.Pod) map[*NodeClaimTemplate]*scheduling.HostPortUsage {
+	nctToOccupiedPorts := map[*NodeClaimTemplate]*scheduling.HostPortUsage{}
+	for _, nct := range nodeClaimTemplates {
+		hostPortUsage := scheduling.NewHostPortUsage()
+		// gather compatible daemon sets for the NodeClaimTemplate of a NodePool
+		for _, pod := range lo.Filter(daemonSetPods, func(p *corev1.Pod, _ int) bool { return isDaemonPodCompatible(nct, p) }) {
+			hostPortUsage.Add(pod, scheduling.GetHostPorts(pod))
+		}
+		if _, ok := nctToOccupiedPorts[nct]; !ok {
```

Comment:
What do we need to do an `ok` check for?

Reply:
Originally had to debug a nil ptr issue and then forgot to take this out.